src/main/java/core/basesyntax/FileWork.java

```diff
@@ -1,14 +1,38 @@
 package core.basesyntax;
 
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Scanner;
+
 /**
  * <p>Дано файл, потрібно прочитати його вміст і вибрати всі слова що починаються на `w`.
  * Результат повернути у вигляді відсортованого масиву (за замовчуванням). Всі слова повинні
  * бути в нижньому регістрі. У випадку якщо таких слів не знайдено повернути пустий масив.
  * Приклад: Width world Wide web
  * Результат: web wide width world</p>
  */
+
 public class FileWork {
+    public static final String CHARACTER = "w";
+
     public String[] readFromFile(String fileName) {
-        return null;
+        File file = new File(fileName);
+        StringBuilder scannedWords = new StringBuilder();
+        try (Scanner scan = new Scanner(file)) {
+            while (scan.hasNext()) {
+                String oneString = scan.next().toLowerCase();
+                if (oneString.startsWith(CHARACTER)) {
+                    scannedWords.append(oneString.replaceAll("\\W", "")).append(" ");
+                }
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Неверно указано имя файла, или его не существует", e);
+        }
+        return scannedWords.length() > 0
+                ? Arrays.stream(scannedWords.toString().trim()
+                .split(" ")).sorted().toArray(String[]::new)
+                : new String[0];
```

Comment:
```suggestion
        return scannedWords.toString().split(" ");
```
this should work, if not - change your code. sorry, didn't notice at first review

Reply:
your option does not work

Reply:
yes, but they won't be sorted. try to use Arrays.sort() with a result of that split method